app/models/events/search.rb

```diff
@@ -51,27 +51,35 @@ def available_months
     end
 
     def query_events
```

Comment:
Having the two search methods feels a little confusing here. I wonder if we really need to enforce the `presence` on month in the model, given it's only picking up on the scenario when someone faffs with the HTML intentionally. If we dropped it then we could revert to a single method:

```
def query_events(limit = RESULTS_PER_TYPE)
  valid ? query_events_api(limit) : {}
end
```

Or if we keep the conditional validation I think we need to make the naming of the use cases clearer as `query_events` and `filter_events` are a bit misleading IMO.

Reply:
Sounds like a good move. I wasn't sure how important the validation on time was so left it.